spec/tests/require.fmf

```diff
@@ -64,12 +72,14 @@ example:
     require:
       - url: https://github.com/beakerlib/openssl
         name: /certgen
+        type: library
```

Comment:
This key is defined with `library` default, but examples add it everywhere, suggesting it's not that optional as it might sound. I'd propose to add an example or two to illustrate the fact it can be omitted. For example (starting with the existing examples, with `type`, plus optional examples at the end):

```yaml
  - |
    # Library from a custom remote git repository
    require:
      - url: https://github.com/beakerlib/openssl
        name: /certgen
        type: library

  - |
    # Library from the local filesystem
    require:
      - path: /tmp/library/openssl
        name: /certgen
        type: library

  - |
    # Use a custom git ref and nick for the library
    require:
      - url: https://github.com/redhat-qe-security/certgen
        ref: devel
        nick: openssl
        name: /certgen
        type: library

  - |
    # Since "library" is the default value for "type", you can omit it when requiring libraries
    require:
      - url: https://github.com/beakerlib/openssl
        name: /certgen

      - path: /tmp/library/openssl
        name: /certgen

      - url: https://github.com/redhat-qe-security/certgen
        ref: devel
        nick: openssl
        name: /certgen
```